### ISSUE TITLE
Micro stuttering on Galaxy S8

### ISSUE THREAD

**Issue by vladone97** [moonlight-stream/moonlight-android#649]
No matter what I try to stream from my computer to my phone, there's always some microstuttering every 4 seconds or so. It doesn't matter if I stream over my home network or over the Internet, or how high or low I set the bitrate. And this doesn't happen on any other device I've tried it on, Android or otherwise. I tested it on my mum's Galaxy A5 2017, my laptop and my PS Vita and they all work fine. I tried both the regular version of Moonlight and the version for rooted devices. I stream with bluetooth turned off, game mode on, no background processes (I disabled background processed from the developer settings) and I even used a wi-fi adapter to create a hotspot so I can connect the phone directly to the computer. Nothing works.

Update: This seems to happen with Remotr Game Streaming as well, so I guess it's safe to say it happens with every game streamign software. I've found that the problem goes away if I turn on airplane mode. But that creates another problem: if I use airplane mode, this means I can't stream over the Internet using mobile data.

**Comment by cgutman** [comment]
You could try disabling location and NFC too. That's most of what else Airplane Mode turns off. See if that helps.

**Comment by vladone97** [comment]
Still no luck. It stutters even with NFC and location services disabled.

**Comment by Minorique** [comment]
Hi all! I really don't know how it works, but... There were microstuttering every 4 seconds on my Galaxy S8 Plus (Exynos). Here is the way to fix it:

1. On your phone go to Settings/Developer options/Running services. Close the "Samsung connect" service. Wait until it dissapears.
2. Then swipe down from the top of the screen, find the location (gps) toggleable icon, and turn it off.

No more microstuttering after that!

**Comment by RobPFarley** [comment]
Still having this issue after disabling Samsung Connect

**Comment by Hamza141** [comment]
Disabling Bluetooth fixed it for me on my Note 8. I was connected to my Gear S3 so I assume communicating with it was causing the stuttering.

**Comment by wanhuz** [comment]
I'm leaving this comment for future peoples that has the same problem as me.

After painful hours of testing, I finally found that disabling SmartThings apps on my Samsung Galaxy Tab 8.0 S-Pen fixed micro-stuttering on Moonlight. To be more specific, disabling SmartThings disables BeaconService, which is common problem people had that cause stutters in Samsung device. Apparently, BeaconService is now packaged inside SmartThings, which causes confusion for myself. 

To disable the app, find it under Settings -> apps -> show system apps (Android 9.0) or disable BeaconService directly under Developer options -> Running services -> SmartThings

I don't know what this service does but from observation, having BeaconService enabled cause rendering frame rate to drop significantly so it cannot catch up with incoming frame rate from host PC, and thus, causing stutters.

**Comment by radugrecu97** [comment]
I want to add a related solution. I've had a lot of network latency spiked every few seconds on my Galaxy Tab S8 regardless of bitrate (although it seemed to happen more often with high bitrate).

The same bitrate works just fine on the same network but on the Steam Deck. Of course this could've been attributed to a better GPU hardware decoding.

BUT, @wanhuz comment made me look into Services on my Tab S8 and I saw "Find my device" service. I disabled it.

I also enabled a developer option that suspends all cached processes, meaning that while Moonlight is running, nothing should be running in the background.

One of these changes fixed the high spikes for me :)
